modules/affile/logproto-file-writer.c

```diff
@@ -45,6 +45,48 @@ typedef struct _LogProtoFileWriter
   struct iovec buffer[0];
 } LogProtoFileWriter;
 
+static inline gboolean
+_flush_partial(LogProtoFileWriter *self, LogProtoStatus *status)
+{
+  /* there is still some data from the previous file writing process */
+
+  gint len = self->partial_len - self->partial_pos;
+  gssize rc = log_transport_write(self->super.transport, self->partial + self->partial_pos, len);
+
+  if (rc > 0 && self->fsync)
+    fsync(self->fd);
+
+  if (rc < 0)
+    {
+      if (errno == EINTR || errno == EAGAIN)
+        {
+          *status = LPS_SUCCESS;
+          return FALSE;
+        }
+
+      log_proto_client_msg_rewind(&self->super);
+      msg_error("I/O error occurred while writing",
+                evt_tag_int("fd", self->super.transport->fd),
+                evt_tag_error(EVT_TAG_OSERROR));
+
+      *status = LPS_ERROR;
+      return FALSE;
+    }
+
+  if (rc != len)
+    {
+      self->partial_pos += rc;
+      *status = LPS_PARTIAL;
+      return FALSE;
+    }
+
+  log_proto_client_msg_ack(&self->super, self->partial_messages);
+  g_free(self->partial);
+  self->partial = NULL;
+  self->partial_messages = 0;
```

Comment:
[optional]

As far as I can see, this was not part of the original code. I guess it is needed, but probably should be introduced in another commit.

Reply:
It's unnecessary, it was just part of my thought process. I can either remove it or commit it separately.

Reply:
I am okay with either, or keeping it like this, too. Let me know if I should merge it this way :)

Reply:
I've extracted it to a separate commit. Thanks.